### PR TITLE
add support for periodic finite difference spaces

### LINKED ISSUE #545: Allow for periodic finite difference meshes.
This would be very helpful for testing our stencils as we don't have to then worry about boundary conditions.

cc: @akshaysridhar @valeriabarra

### PR DESCRIPTION
**Pull request by simonbyrne**
Fixes #545 

Specify `periodic=true` when constructing the `IntervalDomain`.

### CODE REVIEW

File: test/Operators/finitedifference/column.jl [hunk 1]
```diff
@@ -87,6 +87,54 @@ end
     end
 end
 
+
+@testset "Scalar Field FiniteDifferenceSpaces - periodic" begin
```
Comment (valeriabarra) [inline test/Operators/finitedifference/column.jl:91]: Thanks for adding this. Can we also add Mesh and Topology tests, in addition to Operators test?

Comment (simonbyrne) [reply test/Operators/finitedifference/column.jl:91]: I've added Mesh tests. We don't really use any of the Topology properties.

**Review by valeriabarra — APPROVED** [review]
